dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/HibernateTrackedEntityStore.java

```diff
@@ -987,12 +956,7 @@ private String getQueryOrderBy(TrackedEntityQueryParams params, boolean innerOrd
       return "ORDER BY " + StringUtils.join(orderFields, ',') + SPACE;
     }
 
-    if (params.getFilters().entrySet().stream()
-        .noneMatch(qi -> !qi.getValue().isEmpty() && qi.getKey().isUnique())) {
-      return "ORDER BY " + DEFAULT_ORDER + " ";
-    }
-
-    return "";
+    return "ORDER BY " + DEFAULT_ORDER + " ";
```

Comment:
@stian-sandvold Do you know why at some point add the possibility to create a query without default order if there is a filter on an attribute that is unique?

It seems to me that it is safe and it is not impacting performance just to maintain that order by clause

Reply:
I am not sure why this is here right now; I am sure it made sense at some point, but as far as I can tell now; This just means the default sorting order is not always the same, which is not great.

I think always using default, if nothing was explicitly ordered in the params, is the way to go.